include/backend/sycl_backend.h

```diff
@@ -78,6 +105,9 @@ class sycl_backend : public backend {
 
 	system_info& get_system_info(); // mutable system_info is filled by sycl_cuda_backend constructor
 
+	// Enqueues a task on the worker thread corresponding to the given device, and wraps the async_event returned by the task in a delayed_async_event
+	async_event enqueue_device_work(device_id device, std::function<async_event()> work);
```

Comment:
This implementation detail should not be in the header.
Also consider dropping the `std::function` and having a templated functor instead, no need for an allocation here.